mm2src/mm2_bitcoin/keys/src/address/address_builder.rs

```diff
@@ -5,70 +5,110 @@ use {Address, AddressFormat, AddressHashEnum, AddressPrefixes, AddressScriptType
 #[derive(PartialEq)]
 pub enum AddressBuilderOption {
     /// build for pay to pubkey hash output (witness or legacy)
-    BuildPubkeyHash,
+    BuildAsPubkeyHash,
     /// build for pay to script hash output (witness or legacy)
-    BuildScriptHash,
+    BuildAsScriptHash,
 }
 
-/// Builder for Address
-/// Returns Address struct depending on addr_format value checking other properties
+/// Builds Address struct depending on addr_format and validates builder params
 pub struct AddressBuilder {
     /// Coin base58 address prefixes from coin config
-    pub prefixes: NetworkAddressPrefixes,
+    prefixes: NetworkAddressPrefixes,
     /// Segwit addr human readable part
-    pub hrp: Option<String>,
-    /// Public key hash.
-    pub hash: AddressHashEnum,
+    hrp: Option<String>,
+    /// Public key hash
+    hash: AddressHashEnum,
     /// Checksum type
-    pub checksum_type: ChecksumType,
+    checksum_type: ChecksumType,
     /// Address Format
-    pub addr_format: AddressFormat,
+    addr_format: AddressFormat,
+    /// Indicate whether tx output for this address is pubkey hash or script hash
+    build_option: Option<AddressBuilderOption>,
 }
 
 impl AddressBuilder {
-    /// Builds Address with the script type corresponding to p2pkh or p2wpkh
-    pub fn build_as_pkh(&self) -> Result<Address, String> { self.build(AddressBuilderOption::BuildPubkeyHash) }
+    pub fn new(
+        addr_format: AddressFormat,
+        hash: AddressHashEnum,
+        checksum_type: ChecksumType,
+        prefixes: NetworkAddressPrefixes,
+        hrp: Option<String>,
+    ) -> Self {
+        Self {
+            addr_format,
+            hash,
+            checksum_type,
+            prefixes,
+            hrp,
+            build_option: None,
+        }
+    }
+
+    /*pub fn with_hrp(&mut self, hrp: String) -> Self {
+        self.hrp = Some(hrp);
+        self
+    }
+
+    pub fn with_address_prefixes(&mut self, prefixes: NetworkAddressPrefixes) -> Self {
+        self.prefixes = prefixes;
+        self
+    }*/
```

Comment:
I guess you forgot to remove this commented out block.

Reply:
done